.github/workflows/NodeJS.yml

```diff
@@ -236,6 +238,13 @@ jobs:
           node -v
           npm -v
 
+      - name: Setup Ccache
+        uses: hendrikmuhs/ccache-action@main
+        with:
+          key: ${{ github.job }}-${{ matrix.node }}
+          save: ${{ github.ref == 'refs/heads/master' || github.repository != 'duckdb/duckdb' }}
+          variant: sccache
```

Comment:
Any idea why we're using sccache here and ccache everywhere else?

Reply:
I have to say I have no clue about the difference between the two, and haven't experimented (yet?).
In this commit (https://github.com/duckdb/duckdb/pull/5999/commits/e42d4dc798b835cf7e1a88e8aac5adef7e7073e1) I just moved 
I took the original block and moved it all together upwards, since it seemed to me to have marginally more sense, but in the current state NodeJS workflow do not work on my fork (will go back to this) so I have yet to measure whether moving this makes sense.

Thanks for spotting this, could be nice understanding which variant is needed there, will do some checks.

Reply:
Probably places where it shows more are like the LinuxRelease workflow, job 'Linux (64 Bit)', where the build step goes from 13 minutes (https://github.com/carlopi/duckdb/actions/runs/3981348612/jobs/6825028545) to 30 seconds (https://github.com/carlopi/duckdb/actions/runs/4010035135/jobs/6886077180).

Obviously this impact only build steps, but should still be relevant.

Reply:
Maybe we should switch to purely using sccache and having the cache in S3 if we're heading so many issues with cache size? Could even use the cache for local dev then

Reply:
One thing that's nice of the github version is that it play nicely for forks (that in general will not have cloud keys), and it's probably simpler / integrated better (especially if it's true that PR may get cache data from the forks, but haven't experimented on that). But I don't have a strong / well formed opinion.

Reply:
Is sccache running much much faster?

https://github.com/hendrikmuhs/ccache-action#readme mentions:

> Use Sccache for stable Windows support.

What kind of cloud keys do we need for sccache?

Slightly unrelated: Why do we enable all workflows for all branches? The default in other projects is to enable only for the main branch, and for PRs. The downside is that we lose GHA for forks. We could still support *running* the actions only in forks or on our main branch, like this PR is doing for saving.

Reply:
sccache: very much unsure, it's one of the things that will need to be looked at, but I'd say as an independent PR.

select workflows: duckdb/duckdb has a single `master` branch, so this point is partially moot, in the sense that for the main repo workflows are only ever run on master and on PRs, while on forks anyone is free to have as many branches as they prefer, and GH will rate-limit the workflows being run.